components/telemetry-operator/controllers/logparser/controller.go

```diff
@@ -32,33 +31,43 @@ import (
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-// LogParserReconciler reconciles a LogParser object
-type LogParserReconciler struct {
+// Reconciler reconciles a LogParser object
+type Reconciler struct {
 	client.Client
-	Scheme             *runtime.Scheme
-	FluentBitDaemonSet types.NamespacedName
-	Parser             *parserSync.LogParserSyncer
-	DaemonSetUtils     *fluentbit.DaemonSetUtils
+	Scheme                   *runtime.Scheme
+	syncer                   *syncer
+	fluentBitDaemonSetHelper *fluentbit.DaemonSetHelper
 }
 
-func NewLogParserReconciler(client client.Client, scheme *runtime.Scheme, daemonSetConfig parserSync.FluentBitDaemonSetConfig, restartsTotal prometheus.Counter) *LogParserReconciler {
-	var lpr LogParserReconciler
+func NewLogParserReconciler(
+	client client.Client,
+	scheme *runtime.Scheme,
+	fluentBitK8sResources fluentbit.KubernetesResources,
+	restartsTotal prometheus.Counter) *Reconciler {
+	var lpr Reconciler
+
 	lpr.Client = client
 	lpr.Scheme = scheme
```

Comment:
Can you quickly check if `Scheme` and `Client` needs to be public

Reply:
I think `Scheme` can even be removed, `Client` is public because the reconciler itself is a k8s client